code/lib/manager-api/src/modules/channel.ts

```diff
@@ -25,6 +27,11 @@ export const init: ModuleFn<SubAPI, SubState> = ({ provider }) => {
 
       return () => provider.channel.removeListener(type, cb);
     },
+    onServerAction: (type, cb) => {
+      provider.serverChannel.addListener(type, cb);
+
+      return () => provider.serverChannel.removeListener(type, cb);
+    },
```

Comment:
- We should make this API available in the preview also
- We should update the places we listen to `STORY_INDEX_INVALIDATED` ([manager](https://github.com/storybookjs/storybook/blob/01ab4508ce72c72b52a5a28a225400bee3f29846/code/lib/manager-api/src/modules/stories.ts#L566), [preview](https://github.com/storybookjs/storybook/blob/01ab4508ce72c72b52a5a28a225400bee3f29846/code/lib/preview-api/src/modules/preview-web/Preview.tsx#L99)) to use the API.

Reply:
@tmeasday I don't really want to start work combining the serverChannel and normal channel into a single concept at this time.

Looking at:
https://github.com/storybookjs/storybook/blob/c8acbf3db18c8117ef5aa8045a4c6063d9d31fce/code/lib/preview-api/src/modules/preview-web/Preview.tsx#L98-L106

I don't think this really needs a change at this time.
```
this.serverChannel?.on
this.channel.on
```
... that seems fine to me for now.
I could be OK adding a `useServerChannel` hook to `preview-api`, but do you feel like we should add this now, or can that happen when we're in need of it?

Reply:
Ok, cool this seems like a solid step forward for now, for sure!